joern-cli/frontends/php2cpg/src/main/scala/io/joern/php2cpg/passes/PhpSetKnownTypes.scala

```diff
@@ -23,17 +26,20 @@ case class KnownFunction(
   pTypes: Seq[Seq[String]] = Seq.empty
 )
 
+case class PhpSetKnownTypesConfig(knownTypesFilePath: Option[String] = None)
```

Comment:
Feel free to extend this on top of `TypeRecoveryPass` as I am currently doing something similar in `csharpsrc2cpg`

Reply:
Is there an example in the C# frontend that I should reference and match? I don't see it in `master`. 

I use both the `TypeRecoveryParserConfig` and the `PhpSetKnownTypesParserConfig` traits for the php2cpg.Config class in Main.scala; do you prefer that `PhpSetKnownTypesParserConfig` extends directly from `TypeRecoveryParserConfig`, rather than being used along side it as a trait?

Reply:
Nothing in C# yet, but I'm deserializing the object into `TypeMap` which you can find in C#. Right now it's pre-defined, but I'd like to extend it to user-defined paths like this too.

I mean that you can add the `knownTypesFilePath` property to `TypeRecoveryParserConfig` and get rid of `PhpSetKnownTypesParserConfig`.

Reply:
Actually, better yet, keep it and rename it to `TypeStubParserConfig` and then it can be re-used across frontends. It can be placed in the `io.joern.x2cpg` package next to `TypeRecoveryParserConfig` somewhere in `frontends`

Reply:
Got it, thanks! That makes sense.